### PR TITLE
make the choice name optional

### LINKED ISSUE #1311: Make the name optional in skrub.choose_from
We should find a way of guessing name in skrub.choose_from (for instance inspecting the stack to get a meaningful name and adding a number in case of duplicates), and make the name argument optional

### PR DESCRIPTION
**Pull request by jeromedockes**
closes #1311 

The `name` parameter of all choices is now optional. When displaying results if there is no name the slightly shortened repr of the choice is used:

```
>>> from sklearn.datasets import make_classification
>>> from sklearn.preprocessing import StandardScaler
>>> from sklearn.linear_model import LogisticRegression
>>> from sklearn.ensemble import RandomForestClassifier

>>> import skrub

>>> X, y = make_classification()

>>> search = (
...     skrub.X(X)
...     .skb.apply(StandardScaler(with_mean=skrub.choose_bool()))
...     .skb.apply(
...         skrub.choose_from(
...             {
...                 "logistic": LogisticRegression(
...                     C=skrub.choose_float(0.1, 10.0, log=True),
...                     fit_intercept=skrub.choose_bool(),
...                 ),
...                 "rf": RandomForestClassifier(n_estimators=skrub.choose_int(5, 100)),
...             }
...         ),
...         y=skrub.y(y),
...     )
...     .skb.get_randomized_search(fitted=True)
... )
>>> search.results_.iloc[0]
mean_test_score                          0.98
choose_bool()                            True
choose_float(0.1, 10.0, log=True)         NaN
choose_bool()_1                           NaN
choose_int(5, 100)                       36.0
choose_from({'logistic': …, 'rf': …})      rf
Name: 0, dtype: object
```

I noticed plotly does not adjust the space it needs for the tick labels in the parallel coordinate plots, so the names are truncated. However that is also the case for long user-chosen names

screenshot_2025-04-15T11:19:11+02:00

reducing the labelangle helps

screenshot_2025-04-15T11:33:57+02:00

### CODE REVIEW

**Review by Vincent-Maladiere — APPROVED** [review]
LGTM

**Comment by jeromedockes** [comment]
thanks for the review!
